include/engine/guidance/collapse_turns.hpp

```diff
@@ -140,6 +140,8 @@ void combineRouteSteps(RouteStep &step_at_turn_location,
 // alias for suppressing a step, using CombineRouteStep with NoModificationStrategy only
 void suppressStep(RouteStep &step_at_turn_location, RouteStep &step_after_turn_location);
 
+std::vector<RouteStep> suppressSegregated(std::vector<RouteStep> steps);
```

Comment:
I don't really agree with the naming here / use of a dedicated function. Collapse scenarios should make use of the segregated flag instead of checking for length values.

Reply:
Please, clarify in more details. Maybe, the naming is not good, but this function checks for segregated flag, but not the length ... Using the dedicated function at this moment is to minimize changes in current code and don't break existing test logic.